### PR TITLE
Dtype dispatch: add test for dtype check to NumpyBackend

### PR DESCRIPTION
**Pull request by prasunanand**
* Added dtype check for multiple dtypes
* Currently I added a check if `backend == NumpyBackend` before checking for `dtype`

### CODE REVIEW

File: unumpy/tests/test_numpy.py [hunk 1]
```diff
@@ -195,6 +195,9 @@ def test_multiple_output(backend, method, args, kwargs):
     assert all(isinstance(arr, types) for arr in ret)
 
 
+dtypes = [onp.int8, onp.int16, onp.int32, onp.float32, onp.float64]
```
Comment (hameerabbasi) [inline unumpy/tests/test_numpy.py]: Replace these with strings, e.g.
```suggestion
dtypes = ["int8", "int16", "int32", "float32", "float64"]
```

File: unumpy/tests/test_numpy.py [hunk 2]
```diff
@@ -205,12 +208,29 @@ def test_multiple_output(backend, method, args, kwargs):
 )
 def test_array_creation(backend, method, args, kwargs):
     backend, types = backend
-    try:
```
Comment (hameerabbasi) [inline unumpy/tests/test_numpy.py:208]: Why only the NumPy backend?

Comment (prasunanand) [reply unumpy/tests/test_numpy.py:208]: Tests were failing for `xnd_backend` currently.
```
______________ test_array_creation[backend4-zeros-args0-kwargs0] _______________

backend = <module 'unumpy.xnd_backend' from '/home/prasun/dev/labs/unumpy/unumpy/xnd_backend.py'>
method = <uarray multimethod 'zeros'>, args = ((1, 2, 3),)
kwargs = {'dtype': <class 'numpy.int8'>}

    @pytest.mark.parametrize(
        "method, args, kwargs",
        [
            (np.zeros, ((1, 2, 3),), {}),
            (np.ones, ((1, 2, 3),), {}),
            (np.full, ((1, 2, 3), 1.3), {}),
        ],
    )
    def test_array_creation(backend, method, args, kwargs):
        backend, types = backend
        # if backend == NumpyBackend:
        for dtype in dtypes:
            try:
                with ua.set_backend(backend, coerce=True):
                    kwargs["dtype"] = dtype
                    ret = method(*args, **kwargs)
            except ua.BackendNotImplementedError:
                if (
                    backend in FULLY_TESTED_BACKENDS
                    and (backend, method) not in EXCEPTIONS
                ):
                    raise
                pytest.xfail(reason="The backend has no implementation for this ufunc.")
    
            assert isinstance(ret, types)
            print(backend)
>           assert ret.dtype == dtype
E           assert ndt("int8") == <class 'numpy.int8'>
E            +  where ndt("int8") = <[ValueError('The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()') raised in repr()] array object at 0x7f633d4f0350>.dtype

unumpy/tests/test_numpy.py:227: AssertionError
----------------------------- Captured stdout call -----------------------------
<module 'unumpy.xnd_backend' from '/home/prasun/dev/labs/unumpy/unumpy/xnd_backend.py'>
_______________ test_array_creation[backend4-ones-args1-kwargs1] _______________

backend = <module 'unumpy.xnd_backend' from '/home/prasun/dev/labs/unumpy/unumpy/xnd_backend.py'>
method = <uarray multimethod 'ones'>, args = ((1, 2, 3),)
kwargs = {'dtype': <class 'numpy.int8'>}

    @pytest.mark.parametrize(
        "method, args, kwargs",
        [
            (np.zeros, ((1, 2, 3),), {}),
            (np.ones, ((1, 2, 3),), {}),
            (np.full, ((1, 2, 3), 1.3), {}),
        ],
    )
    def test_array_creation(backend, method, args, kwargs):
        backend, types = backend
        # if backend == NumpyBackend:
        for dtype in dtypes:
            try:
                with ua.set_backend(backend, coerce=True):
                    kwargs["dtype"] = dtype
                    ret = method(*args, **kwargs)
            except ua.BackendNotImplementedError:
                if (
                    backend in FULLY_TESTED_BACKENDS
                    and (backend, method) not in EXCEPTIONS
                ):
                    raise
                pytest.xfail(reason="The backend has no implementation for this ufunc.")
    
            assert isinstance(ret, types)
            print(backend)
>           assert ret.dtype == dtype
E           assert ndt("int8") == <class 'numpy.int8'>
E            +  where ndt("int8") = <[ValueError('The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()') raised in repr()] array object at 0x7f633d4f0a50>.dtype

unumpy/tests/test_numpy.py:227: AssertionError
----------------------------- Captured stdout call -----------------------------
<module 'unumpy.xnd_backend' from '/home/prasun/dev/labs/unumpy/unumpy/xnd_backend.py'>
_______________ test_array_creation[backend4-full-args2-kwargs2] _______________

backend = <module 'unumpy.xnd_backend' from '/home/prasun/dev/labs/unumpy/unumpy/xnd_backend.py'>
method = <uarray multimethod 'full'>, args = ((1, 2, 3), 1.3)
kwargs = {'dtype': <class 'numpy.int8'>}

    @pytest.mark.parametrize(
        "method, args, kwargs",
        [
            (np.zeros, ((1, 2, 3),), {}),
            (np.ones, ((1, 2, 3),), {}),
            (np.full, ((1, 2, 3), 1.3), {}),
        ],
    )
    def test_array_creation(backend, method, args, kwargs):
        backend, types = backend
        # if backend == NumpyBackend:
        for dtype in dtypes:
            try:
                with ua.set_backend(backend, coerce=True):
                    kwargs["dtype"] = dtype
                    ret = method(*args, **kwargs)
            except ua.BackendNotImplementedError:
                if (
                    backend in FULLY_TESTED_BACKENDS
                    and (backend, method) not in EXCEPTIONS
                ):
                    raise
                pytest.xfail(reason="The backend has no implementation for this ufunc.")
    
            assert isinstance(ret, types)
            print(backend)
>           assert ret.dtype == dtype
E           assert ndt("int8") == <class 'numpy.int8'>
E            +  where ndt("int8") = <[ValueError('The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()') raised in repr()] array object at 0x7f633d4f0f50>.dtype

unumpy/tests/test_numpy.py:227: AssertionError
----------------------------- Captured stdout call -----------------------------
<module 'unumpy.xnd_backend' from '/home/prasun/dev/labs/unumpy/unumpy/xnd_backend.py'>
```

Comment (prasunanand) [reply unumpy/tests/test_numpy.py:208]: Looking into it. Will modify the PR accordingly.

Comment (prasunanand) [reply unumpy/tests/test_numpy.py:208]: @hameerabbasi , Please review again :)

**Review by hameerabbasi — APPROVED** [review]

**Comment by hameerabbasi** [comment]
@prasunanand Is this ready to go? If so, please mark it Ready for review.

**Comment by prasunanand** [comment]
I need to add more tests(ufunc). After that I will mark it Ready for Review. I needed to know if my approach was right.

**Review by hameerabbasi — APPROVED** [review]

**Pull request merged**